src/wp-includes/option.php

```diff
@@ -827,7 +822,15 @@ function update_option( $option, $value, $autoload = null ) {
 	);
 
 	if ( null !== $autoload ) {
-		$update_args['autoload'] = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+		$autoload                = determine_option_autoload_value( $option, $serialized_value, $autoload );
+		$update_args['autoload'] = $autoload;
+	} else {
+		$raw_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s LIMIT 1", $option ) );
+		$allow_values = array( 'default-yes', 'default-no' );
+		if ( in_array( $raw_autoload, $allow_values, true ) ) {
+			$autoload                = determine_option_autoload_value( $option, $serialized_value, $autoload );
+			$update_args['autoload'] = $autoload;
```

Comment:
Better to only update the value if it's actually different. This also has implications on the if clauses further below in the function, where right now the `$update_args['autoload']` value would almost always be set. We should probably avoid that and only set it when needed.
```suggestion
			$autoload = determine_option_autoload_value( $option, $serialized_value, $autoload );
			if ( $autoload !== $raw_autoload ) {
				$update_args['autoload'] = $autoload;
			}
```